src/libs/kata-sys-util/src/k8s.rs

```diff
@@ -49,23 +51,64 @@ pub fn is_host_empty_dir(path: &str) -> bool {
     false
 }
 
-// update_ephemeral_storage_type sets the mount type to 'ephemeral'
+pub fn patchup_container_storage_type(oci_spec: &mut Spec) -> Result<()> {
+    for m in oci_spec.mounts.iter_mut() {
+        patchup_ephemeral_volume_type(m)?;
+        patchup_direct_volume_type(m)?;
+    }
+
+    Ok(())
+}
+
+// patchup_ephemeral_volume_type sets the mount type to 'ephemeral'
 // if the mount source path is provisioned by k8s for ephemeral storage.
 // For the given pod ephemeral volume is created only once
 // backed by tmpfs inside the VM. For successive containers
 // of the same pod the already existing volume is reused.
-pub fn update_ephemeral_storage_type(oci_spec: &mut Spec) {
-    for m in oci_spec.mounts.iter_mut() {
-        if mount::is_kata_guest_mount_volume(&m.r#type) {
-            continue;
-        }
+fn patchup_ephemeral_volume_type(mount: &mut Mount) -> Result<()> {
+    if mount::is_kata_guest_mount_volume(&mount.r#type) {
+        return Ok(());
+    }
 
-        if is_ephemeral_volume(&m.source) {
-            m.r#type = String::from(mount::KATA_EPHEMERAL_VOLUME_TYPE);
-        } else if is_host_empty_dir(&m.source) {
-            // FIXME support disable_guest_empty_dir
-            // https://github.com/kata-containers/kata-containers/blob/02a51e75a7e0c6fce5e8abe3b991eeac87e09645/src/runtime/pkg/katautils/create.go#L105
-            m.r#type = String::from(mount::KATA_HOST_DIR_VOLUME_TYPE);
+    if is_ephemeral_volume(&mount.source) {
+        mount.r#type = String::from(mount::KATA_EPHEMERAL_VOLUME_TYPE);
+    } else if is_host_empty_dir(&mount.source) {
+        // FIXME support disable_guest_empty_dir
+        // https://github.com/kata-containers/kata-containers/blob/02a51e75a7e0c6fce5e8abe3b991eeac87e09645/src/runtime/pkg/katautils/create.go#L105
+        mount.r#type = String::from(mount::KATA_HOST_DIR_VOLUME_TYPE);
+    }
+
+    Ok(())
+}
+
+// patchup_direct_volume_type trys to patch up the created direct volume mount for the container.
+// Note: Not all mounts require patching; our focus is solely on direct volumes.
+fn patchup_direct_volume_type(mount: &mut Mount) -> Result<()> {
+    // Only handle bind Mount
+    if mount.r#type != "bind" {
+        return Ok(());
+    }
+
+    match get_volume_mount_info(&mount.source) {
+        Ok(m) => {
+            mount.r#type = m.volume_type;
+            mount.options = vec!["rbind".to_owned(), "rw".to_owned()];
+        }
+        Err(e) => {
+            if let Some(&error_kind) = e.downcast_ref::<std::io::ErrorKind>() {
+                match error_kind {
+                    std::io::ErrorKind::NotFound => return Ok(()),
```

Comment:
In my opinion, We should leave some comments here to explain why we return Ok when 404 error occurred

Reply:
fixed, comments added to explain it.